### PR TITLE
Ab#144478 update seafarer profile information

### PR DESCRIPTION
**Pull request by tc-chris**

### CODE REVIEW

File: src/CSF.SRDashboard.Client/Services/CommonService.cs [hunk 1]
```diff
@@ -3,16 +3,67 @@
...
+        public long? GetUnixTimeFromDateTime(DateTime? given_date)
+        {
+            long? unix_time = null;
+            if (given_date != null)
+            {
+                DateTime date1970 = new DateTime(1970, 1, 1);
+                var timeSpan = (DateTime)given_date - date1970;
+                unix_time = Convert.ToInt64(timeSpan.TotalMilliseconds);
+            }
+
```
Comment (TC-Cory) [inline src/CSF.SRDashboard.Client/Services/CommonService.cs:32]:
```suggestion
```

File: src/CSF.SRDashboard.Client/Services/GatewayService.cs [hunk 2]
```diff
@@ -237,70 +261,161 @@ public async Task<List<MpdisCertificateDto>> GetCertificatesByCdnFromMpdisAsync(
...
+                                { continue; }
+
+                                if (sourcePropertyInfo.PropertyType == typeof(DateTime) && (DateTime)fieldValue == DateTime.MinValue)
+                                { continue; }
+
+                                if (destinationPropertyInfo.CanWrite)
+                                {
+                                    destinationPropertyInfo.SetValue(destinationObject, fieldValue, null);   // sets the corresponding property value of destinationObject to vesselFieldValue.
+                                }
+
```
Comment (TC-Cory) [inline src/CSF.SRDashboard.Client/Services/GatewayService.cs:380]:
```suggestion
```

**Review by TC-Cory — COMMENTED** [review]
All looks good, couple changes. No feedback on Validation if items are removed (IE - Delete First name and save)

File: src/CSF.SRDashboard.Client/Components/Seafarer/SeafarerProfileComponent.razor [hunk 3]
```diff
@@ -101,14 +91,11 @@
...
                                     }
 
-                                    @*@if (GenderSelections.Where(x => x.Id == this.SeafarerProfile.Applicant.FullGender || x.Text == this.SeafarerProfile.Applicant.FullGender).SingleOrDefault() == null)
-                                        {
-                                            <p class="margin-bottom-2rem">@Constants.NotAvailable</p>
-                                        }*@
                                 }
                                 else
                                 {
                                     <FormInputSelect Required="required" IsRequired="true" Id="Gender" Label="@Localizer["Gender"]" SelectLabel="@Localizer["Gender"]" ShowDefaultOption="false" @bind-Value="@SeafarerProfile.Applicant.FullGender" ValidationFor="@(() => SeafarerProfile.Applicant.FullGender)">
```
[the remark this answers is not in the document]
Comment (tc-chris) [reply src/CSF.SRDashboard.Client/Components/Seafarer/SeafarerProfileComponent.razor:97]: Fixed this

**Review by TC-Cory — APPROVED** [review]

**Pull request merged**